### ISSUE TITLE
Ex. 1.5a "pred_err is too big"

### ISSUE THREAD

**Issue by joangog** [sanderbroos/MachineLearning1#1]
In ex. 1.5a (I did it) the calculated prediction error is too big (more than 0.001) and throws an exception. I'm not sure if that means it's wrong but when I change M and reg to the optimal values I calculated at 1.5b (that is M=2 and reg=4.5399929762484875e-05), and not the ones the teacher provided for the example (that is M=2 and reg=0.1), the error is fine (less than 0.001). Maybe the error is too big on purpose? We should discuss it.

**Comment by sanderbroos** [comment]
No I think something is going wrong with the prediction from my function.. Since it's predicting a curve of order 2 and using M=2, the error should be close to zero. I also see the problem when plotting the prediction. I'll take a look if I can find the issue.

**Comment by sanderbroos** [comment]
Yeah we should probably discuss it with the TA. I can't find any mistakes in the code, so maybe the error is indeed that big on purpose? Although it would be weird if the optimal "reg" was that low. I believe setting reg to 0 actually gets an even lower error, which is strange, since then it just becomes plain polynomial regression. But maybe that's logical since the function that is fitted is itself a polynomial. I have no idea 😅

**Comment by joangog** [comment]
@sanderbroos I'll look into it. If you can, in 1.5a check the error function I'm using and tell if that's ok. It isn't mentioned explicitly which one I should use but I assumed it's the one from 1.4 since this is the one we are trying to minimize. Maybe there's something wrong there?

**Comment by sanderbroos** [comment]
The error function looks good to me! I think the problem lies in the predicted fit function, since when I plot it, it's very clearly not lined up very well with t_valid.
![download](https://user-images.githubusercontent.com/70587522/189520390-5b85df1a-5601-4a5e-88b5-b8b5e1e14097.png)
